### PR TITLE
Test with coverage

### PR DESCRIPTION
**Pull request by Stevenjin8**

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -6,7 +6,7 @@ ifeq ($(TLS_MODE), boring)
 endif
 
 test:
-	RUST_BACKTRACE=1 cargo test --benches --tests --bins $(FEATURES)
+	FEATURES=$(FEATURES) ./scripts/test-with-coverage.sh
```
Comment (howardjohn) [inline Makefile.core.mk:9]: can we keep this as a special `coverage` target? Most ofthe time if I just want to test I don't want to waste all the effort and complexity of coverage.

And also not run that in presubmit.

File: scripts/test-with-coverage.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
```
Comment (bleggett) [inline scripts/test-with-coverage.sh:1]:
```suggestion
#!/usr/bin/env sh
```

nit - can we stick to posix sh please, unless we really need a bashism here.

File: scripts/test-with-coverage.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# Clean directory
+rm -rf "$profdata" "$profiles"
+mkdir -p "$profiles"
+
+echo "Running tests"
+cargo test --benches --tests --bins $FEATURES
+
+# Merge profraw data
+echo "Merging profraw files in $profiles to $profdata"
+llvm-profdata merge -sparse $(find "$profiles" -name '*.profraw') -o $profdata
```
Comment (bleggett) [inline scripts/test-with-coverage.sh:43]: `llvm` is not a hard dep of  `rust`, is it? Can we add a check in the makefile that this binary exists in $PATH before running the script?

**Review by bleggett — COMMENTED** [review]
Mostly nits, lgtm

File: scripts/test-with-coverage.sh (hunk 3 above)
Comment (Stevenjin8) [reply scripts/test-with-coverage.sh:43]: My understanding the rust very much depends on llvm. Though its not clear whether having rust means having the whole llvm toolchain.

Comment (bleggett) [reply scripts/test-with-coverage.sh:43]: Oh my bad, it is. (Well, it doesn't have to be, but is in the default builds).

Nevermind then, probably safe to assume.

Comment (bleggett) [reply scripts/test-with-coverage.sh:43]: Yeah on fedora for instance the binary tools appear to be separate/optional. But it's not a big deal.

Comment (Stevenjin8) [reply scripts/test-with-coverage.sh:43]: Also, the test and coverage targets are different

File: scripts/test-with-coverage.sh (hunk 2 above)
Comment (dhawton) [reply scripts/test-with-coverage.sh]: Most scripts around Istio are bash, so probably for consistency sake it should be bash. Doesn't matter too much either way.

Comment (bleggett) [reply scripts/test-with-coverage.sh]: Yep that's fair. I've encountered bugs using scripts that assume bash in environments that don't have bash. And if the script doesn't actually need bash, it's generally pointless and also less compatible to require it.

**Review by bleggett — APPROVED** [review]
TY!

**Review by dhawton — APPROVED** [review]

**Pull request merged**